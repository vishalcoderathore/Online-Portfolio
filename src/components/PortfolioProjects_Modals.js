import _ from 'lodash';
import React from 'react';

const PortfolioProjects_Modals = () => {
  const modalDetails = [
    {
      id: 'project-1',
      labelId: 'project-1-Label',
      title: 'iFeedback',
      mutedText: 'Create Surveys for your Clients',
      href: 'https://ifeedback-emails.herokuapp.com',
      projectInfo: [
        'React App used to create Surveys and send them to single/multiple email addresses.',
        'App uses SendGrid API to create emails, Stripe API to mock payments which are required to purchase credits. These credits are used to create Surveys.',
        'Track all sent Surveys and monitor responses.'
      ],
      appStack: [
        'Axios',
        'Express',
        'Font Awesome',
        'Heroku',
        'Google OAuth',
        'Materialize CSS',
        'Mongo DB',
        'React.JS',
        'Redux State Management',
        'SendGrid API',
        'Stripe Payment API',
        'App is developed with Prod and Dev environments'
      ]
    },
    {
      id: 'project-2',
      labelId: 'project-2-Label',
      title: 'Recipe Finder',
      mutedText: 'Search your favorite recipes online',
      href: 'http://forkify-search-recipes.herokuapp.com/',
      projectInfo: [
        'Web App using JavaScript to search your favorite recipes online. You can search for recipes by typing a query to get a list of items that use the query item, select a recipe to display the details of it.'
      ],
      appStack: ['Axios', 'Babel', 'Express', 'Heroku', 'Node.js', 'Webpack']
    },
    {
      id: 'project-3',
      labelId: 'project-3-Label',
      title: 'Expensify',
      mutedText: 'Track your expenses anytime anywhere',
      href: 'http://track-expenses.herokuapp.com',
      projectInfo: [
        'Web App using React.JS that is used to track your expenses. You can add, view, modify and remove your expenses.',
        'Login using your Google Account. Click on Add Expense to add any new expense. Click on existing expese to either Update or Remove the expense. Click the app logo to reach the dashboard page.'
      ],
      appStack: [
        'Babel',
        'Enzyme',
        'Express',
        'Firebase (database & authtication)',
        'Font Awesome',
        'Heroku',
        'Jest',
        'Node.js',
        'Redux',
        'Webpack'
      ]
    },
    {
      id: 'project-4',
      labelId: 'project-4-Label',
      title: 'Portfolio',
      mutedText: 'You are currently viewing this project',
      href: 'http://track-expenses.herokuapp.com',
      projectInfo: ['Online portfolio showcasing my skills and experience.'],
      appStack: [
        'Babel',
        'Enzyme',
        'Express',
        'Font Awesome',
        'React',
        'AWS - IAM',
        'AWS - Route 53 & DNS',
        'AWS - CloudFront & SSL',
        'AWS - CodeBuild & Code Pipeline',
        'AWS - Lambda Functions'
      ]
    }
  ];

  const listModals = _.map(
    modalDetails,
    ({ id, labelId, title, mutedText, href, projectInfo, appStack }) => {
      return (
        <div
          key={id}
          className='modal fade'
          id={id}
          tabIndex='-1'
          role='dialog'
          aria-labelledby={labelId}
          aria-hidden='true'>
          <div className='modal-dialog' role='document'>
            <div className='modal-content'>
              <div className='modal-header'>
                <h5 className='modal-title' id={labelId}>
                  {title}
                </h5>
                <button
                  type='button'
                  className='close'
                  data-dismiss='modal'
                  aria-label='Close'>
                  <span aria-hidden='true'>&times;</span>
                </button>
              </div>
              <div className='modal-body text-left'>
                <div className='row'>
                  <div className='col-9'>
                    {' '}
                    <p className='text-muted'>{mutedText}</p>
                  </div>
                  <div className='col-3'>
                    <a
                      href={href}
                      target='_blank'
                      className='btn btn-outline-info'>
                      Demo
                    </a>
                  </div>
                </div>
                <React.Fragment>
                  {_.map(projectInfo, (data, index) => (
                    <p key={index}>{data}</p>
                  ))}
                </React.Fragment>
                <p>App is built using the following tech stack :</p>
                <ul>
                  {_.map(appStack, (data, index) => (
                    <li key={index}>{data}</li>
                  ))}
                </ul>
              </div>
            </div>
          </div>
        </div>
      );
    }
  );
  return <div>{listModals}</div>;
};

export default PortfolioProjects_Modals;
