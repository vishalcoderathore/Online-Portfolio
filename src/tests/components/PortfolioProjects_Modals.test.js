import React from 'react';
import { shallow } from 'enzyme';
import PortfolioProjects_Modals from '../../components/PortfolioProjects_Modals';

test('should render portfolio work correctly', () => {
    const wrapper = shallow(<PortfolioProjects_Modals />);
    expect(wrapper).toMatchSnapshot();
});