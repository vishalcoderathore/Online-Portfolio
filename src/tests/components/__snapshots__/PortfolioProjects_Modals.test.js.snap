// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render portfolio work correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="project-1-Label"
    className="modal fade"
    id="project-1"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="project-1-Label"
          >
            Expensify
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body text-left"
        >
          <div
            className="row"
          >
            <div
              className="col-9"
            >
               
              <p
                className="text-muted"
              >
                Track your expenses anytime anywhere.
              </p>
            </div>
            <div
              className="col-3"
            >
              <a
                className="btn btn-outline-info"
                href="http://track-expenses.herokuapp.com"
                target="_blank"
              >
                Demo
              </a>
            </div>
          </div>
          <p>
            Web App using React.JS that is used to track your expenses. You can add, view, modify and remove your expenses.
          </p>
          <p>
            Login using your Google Account. Click on Add Expense to add any new expense. Click on existing expese to either Update or Remove the expense. Click the app logo to reach the dashboard page.
          </p>
          <p>
            App is built using the following tech stack :
          </p>
          <ul>
            <li>
              Babel
            </li>
            <li>
              Enzyme
            </li>
            <li>
              Express
            </li>
            <li>
              Firebase (database and authentication)
            </li>
            <li>
              Font Awesome
            </li>
            <li>
              Heroku
            </li>
            <li>
              Jest
            </li>
            <li>
              Live Server
            </li>
            <li>
              Moment
            </li>
            <li>
              Normalize.CSS
            </li>
            <li>
              Redux
            </li>
            <li>
              SCSS
            </li>
            <li>
              Webpack
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="project-2-Label"
    className="modal fade"
    id="project-2"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="project-1-Label"
          >
            Product Marketing Site
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body text-left"
        >
          <div
            className="row"
          >
            <div
              className="col-9"
            >
               
              <p
                className="text-muted"
              >
                Site showcasing a product.
              </p>
            </div>
            <div
              className="col-3"
            >
              <a
                className="btn btn-outline-info"
                href="http://apple-product.s3-website-us-east-1.amazonaws.com/index.html"
                target="_blank"
              >
                Demo
              </a>
            </div>
          </div>
          <p>
            Multi page sample website showcasing mobile phones and sample plans.
          </p>
          <p>
            App is built using the following tech stack :
          </p>
          <ul>
            <li>
              Bootstrap 4
            </li>
            <li>
              Gulp
            </li>
            <li>
              JQuery
            </li>
            <li>
              Popper.JS
            </li>
            <li>
              SASS
            </li>
            <li>
              Font Awesome
            </li>
            <li>
              Amazon Web Services (AWS) S3
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="project-3-Label"
    className="modal fade"
    id="project-3"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="project-1-Label"
          >
            Indecision
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body text-left"
        >
          <div
            className="row"
          >
            <div
              className="col-9"
            >
               
              <p
                className="text-muted"
              >
                Web App that makes decisions for you. You can add, view, and remove your tasks/options.
              </p>
            </div>
            <div
              className="col-3"
            >
              <a
                className="btn btn-outline-info"
                href="http://indecisionapp.herokuapp.com/"
                target="_blank"
              >
                Demo
              </a>
            </div>
          </div>
          <p>
            Basic App that takes in user inputs and stores in an state array. Then prints out an element in a random fashion.
          </p>
          <p>
            Click on Add Option to add a new option. You can add multiple options or just stick to one. Upon addition of an option, the "What should I do?" button gets unlocked and becomes clickable. When the button is clicked, the app would choose an option for you from the provided options.You can remove "a" option or can also remove all the options at once. The app is also responsive and so would run on your mobile devices as well.
          </p>
          <p>
            App is built using the following tech stack :
          </p>
          <ul>
            <li>
              Babel
            </li>
            <li>
              Express
            </li>
            <li>
              React v15.6.1
            </li>
            <li>
              Webpack v3.1.0
            </li>
            <li>
              Heroku
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="project-4-Label"
    className="modal fade"
    id="project-4"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="project-1-Label"
          >
            My Portfolio
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body text-left"
        >
          <div
            className="row"
          >
            <div
              className="col-9"
            >
               
              <p
                className="text-muted"
              >
                You are currently viewing this project
              </p>
            </div>
          </div>
          <p>
            Online portfolio showcasing my skills and experience.
          </p>
          <p>
            Website is built using the following tech stack :
          </p>
          <ul>
            <li>
              Babel
            </li>
            <li>
              Express
            </li>
            <li>
              Enzyme
            </li>
            <li>
              Font Awesome
            </li>
            <li>
              React v16
            </li>
            <li>
              SASS
            </li>
            <li>
              Webpack v3.1.0
            </li>
            <li>
              Amazon Web Services - IAM
            </li>
            <li>
              Amazon Web Services - Route 53 & DNS
            </li>
            <li>
              Amazon Web Services - CloudFront & SSL
            </li>
            <li>
              Amazon Web Services - CodeBuild & CodePipeline
            </li>
            <li>
              Amazon Web Services - AWS Lambda
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
